### PR TITLE
Finish sbs smk migration

### PR DESCRIPTION
**Pull request by roshankern**
In this PR, we finish the migration of the snakemake code for SBS.

@mat10d this is ready for review!

### CODE REVIEW

File: workflow/lib/shared/external/tifffile.py
[the remark this answers is not in the document]
Comment (mat10d) [reply workflow/lib/shared/external/tifffile.py:1]: We should look to deprecate this. I think we want to see how we can get by using skimage.imread (and not read_stack and tifffile) in the phenotyping step, and then seek to get rid of this

File: workflow/lib/shared/file_utils.py [hunk 1]
```diff
@@ -26,3 +35,121 @@ def get_filename(data_location: dict, info_type: str, file_type: str) -> str:
     # Construct filename by combining components with info_type and file_type
     filename = f"{well_str}{tile_str}{cycle_str}__{info_type}.{file_type}"
     return filename
+
+
+def read_stack(filename, copy=True, maxworkers=None, fix_axes=False):
```
Comment (mat10d) [inline workflow/lib/shared/file_utils.py:40]: As above, let's try to deprecate this in the near future. Will take some digging of why we need read_stack, but I am sure you can figure that out. Not a priority for now

File: workflow/rules/sbs_process.smk [hunk 2]
```diff
@@ -137,3 +137,168 @@ rule apply_ic_field:
...
+        lambda wildcards: expand(
+            SBS_PROCESS_FP
+            / "tsvs"
+            / get_filename(
+                {"well": wildcards.well, "tile": "{tile}"},
+                "minimal_phenotype_info",
+                "tsv",
+            ),
+            tile=SBS_TILES,
+        ),
```
Comment (mat10d) [inline workflow/rules/sbs_process.smk:298]: Unfortunately, we use the wording "merge" as a step in the pipeline. So I would start changing all of these and their underlying functions to "combine"

eg combine_minimal_phenotype_info

File: workflow/scripts/shared/merge_dfs.py
Comment (mat10d) [inline workflow/scripts/shared/merge_dfs.py:1]: Per previous comment, "combine_dfs.py"

**Review by mat10d — APPROVED** [review]
Overall looks good. I'm not too happy with the need to use read_stack, but let's get through all the image processing aspects of the pipeline (phenotyping) and circle back to this. Some minor comments, once resolved, good to go!

File: workflow/lib/shared/external/tifffile.py
[the remark this answers is not in the document]
Comment (roshankern) [reply workflow/lib/shared/external/tifffile.py:1]: Unfortunately we do need to use it for the max_filtered step (as we discussed in Slack). But yes, let's circle back to this once we finish image processing.

File: workflow/rules/sbs_process.smk (hunk 2 above)
Comment (roshankern) [reply workflow/rules/sbs_process.smk:298]: Ah, good point. Done in https://github.com/cheeseman-lab/brieflow/pull/8/commits/3f34212f7984650a2ff4e43cae8f7648e89cdfc0

File: workflow/scripts/shared/merge_dfs.py
Comment (roshankern) [reply workflow/scripts/shared/merge_dfs.py:1]: Done in https://github.com/cheeseman-lab/brieflow/pull/8/commits/3f34212f7984650a2ff4e43cae8f7648e89cdfc0

File: workflow/lib/sbs_process/call_reads.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+    # Map the index to the corresponding base symbol
+    calls = np.array(list(bases))[calls]
+
+    # Combine the base symbols for each cycle to form the called bases for each read
+    return ["".join(x) for x in calls]
+
+
+def quality(X):
+    """Calculate quality scores based on the intensity values.
```
[the remark this answers is not in the document]
Comment (roshankern) [reply workflow/lib/sbs_process/call_reads.py]: Thanks for sharing. Done in https://github.com/cheeseman-lab/brieflow/pull/8/commits/8a5a5235892935d37296a1501fb6672a1da09153

File: workflow/lib/sbs_process/segment_cellpose.py [hunk 4]
```diff
@@ -0,0 +1,447 @@
+"""Cellpose-based Image Segmentation!
+
+This module provides functions for segmenting microscopy images using the Cellpose algorithm
```
[the remark this answers is not in the document]
Comment (mat10d) [reply workflow/lib/sbs_process/segment_cellpose.py:3]: I just didn't see the list based approach in the other docstrings.

[the remark this answers is not in the document]
Comment (roshankern) [reply workflow/lib/sbs_process/segment_cellpose.py:3]: As discussed in Slack, the annotation with a list of functions makes sense in some contexts (like this one).

**Pull request merged**